File: Models/Medico.cs
using System.ComponentModel.DataAnnotations;

namespace ilis.Models
{
    public class Medico
    {
        public int Id { get; set; }

        [MaxLength(50)]
        public string Nome { get; set; }

        [MaxLength(50)]
        public string? Endereco { get; set; }

        [MaxLength(50)]
        public string? Bairro { get; set; }

        [MaxLength(9)]
        public string? Cep { get; set; }

        [MaxLength(50)]
        public string? Cidade { get; set; }

        [MaxLength(50)]
        public string? Telefone { get; set; }

        public int? TipoConselho { get; set; }

        [MaxLength(10)]
        public string? CRM { get; set; }

        [Required, MaxLength(1)]
        public string? Sexo { get; set; }

        [MaxLength(1)]
        public string? Porc { get; set; }

        [Required]
        public decimal Interno { get; set; }

        [Required]
        public decimal Externo { get; set; }

        [MaxLength(20)]
        public string? CodUnimed { get; set; }

        [MaxLength(2)]
        public string? VinUnimed { get; set; }

        [MaxLength(3)]
        public string? EspUnimed { get; set; }

        [MaxLength(20)]
        public string? CodSUS { get; set; }

        [MaxLength(20)]
        public string? CBOS { get; set; }

        [MaxLength(10)]
        public string? SenhaInternet { get; set; }

        [MaxLength(1)]
        public string? Estrangeiro { get; set; }

        [MaxLength(1)]
        public string? StAtivo { get; set; }
    }
}

File: wwwroot/js/medico.js
$(document).ready(function () {
    $('#Medico').on('keypress', function (e) {
        if (e.which == 13) { // Enter key pressed
            e.preventDefault();
            var searchTerm = $(this).val();
            $.get('/api/medicos', { nome: searchTerm }, function (data) {
                var results = $('#medico-search-results');
                results.empty();
                data.forEach(function (medico) {
                    results.append('<li><a href="#" class="select-medico" data-nome="' + medico.crm + ' - ' + medico.nome + '">' + 'Nome: ' + medico.nome + ' CRM: ' + medico.crm + '</a></li>');
                });
                $('#medicoModal').modal('show');
            });
        }
    });

    $(document).on('click', '.select-medico', function (e) {
        e.preventDefault();
        var nome = $(this).data('nome');
        $('#Medico').val(nome);
        $('#medicoModal').modal('hide');
    });

    $('#saveMedicoBtn').on('click', function () {
        var medicoData = {
            Nome: $('#NomeMedico').val(),
            Endereco: $('#EnderecoMedico').val(),
            Bairro: $('#BairroMedico').val(),
            CEP: $('#CepMedico').val(),
            Cidade: $('#CidadeMedico').val(),
            Telefone: $('#TelefoneMedico').val(),
            TipoConselho: $('#TipoConselhoMedico').val(),
            CRM: $('#CRMMedico').val(),
            Sexo: $('#SexoMedico').val(),
            Porc: $('#PorcMedico').val(),
            Interno: $('#InternoMedico').val(),
            Externo: $('#ExternoMedico').val(),
            CodUnimed: $('#CodUnimedMedico').val(),
            VinUnimed: $('#VinUnimedMedico').val(),
            EspUnimed: $('#EspUnimedMedico').val(),
            CodSUS: $('#CodSUSMedico').val(),
            CBOS: $('#cBOSMedico').val(),
            SenhaInternet: $('#SenhaInternetMedico').val(),
            Estrangeiro: $('#EstrangeiroMedico').val(),
            StAtivo: $('#stAtivoMedico').val()
        };
        alert(Sexo);
        $.ajax({
            url: '/api/medicos',
            type: 'POST',
            contentType: 'application/json',
            data: JSON.stringify(medicoData),
            success: function (response) {
                alert('Medico salvo com sucesso!');
                $('#cadastroMedicoModal').modal('hide');
            },
            error: function (xhr) {
                var errors = xhr.responseJSON.errors;
                var errorMessage = "Erro ao salvar medico:\n";
                for (var key in errors) {
                    if (errors.hasOwnProperty(key)) {
                        errorMessage += errors[key] + "\n";
                    }
                }
                alert(errorMessage);
            }
        });
    });
});